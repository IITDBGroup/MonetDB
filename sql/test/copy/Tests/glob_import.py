import os
import sys
try:
    from MonetDBtesting import process
except ImportError:
    import process

server = process.server(stdin=process.PIPE, stdout=process.PIPE, stderr=process.PIPE)
client = process.client('sql', server=server, stdin=process.PIPE, stdout=process.PIPE, stderr=process.PIPE)
cout, cerr = client.communicate("""
start transaction;
create table "ratings" (movie_id BIGINT, customer_id BIGINT, rating TINYINT, rating_date DATE);
copy into "ratings" from '%s' using delimiters ',','\n';
copy into "ratings" from '%s' using delimiters ',','\n';
copy into "ratings" from '%s' using delimiters ',','\n';
copy into "ratings" from '%s' using delimiters ',','\n';
copy into "ratings" from '%s' using delimiters ',','\n'; --error
rollback;
""" % (os.getenv("TSTDATAPATH")+"/netflix_data/ratings_sample_0.csv", os.getenv("TSTDATAPATH")+"/netflix_data/ratings_sample_1.csv", os.getenv("TSTDATAPATH")+"/netflix_data/ratings_sample_?.csv", os.getenv("TSTDATAPATH")+"/netflix_data/ratings_sample_*", os.getenv("TSTDATAPATH")+"/netflix_data/idontexist*"))

sout, serr = server.communicate()
sys.stdout.write(sout)
sys.stderr.write(serr)

sys.stdout.write(cout)
sys.stderr.write(cerr)
