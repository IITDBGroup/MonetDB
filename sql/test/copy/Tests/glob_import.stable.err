stderr of test 'glob_import` in directory 'sql/test/copy` itself:


# 10:27:34 >  
# 10:27:34 >  "/usr/bin/python3" "glob_import.py" "glob_import"
# 10:27:34 >  

# builtin opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-globimport/BUILD/var/monetdb5/dbfarm/demo
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_ipv6 = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 37762
# cmdline opt 	mapi_usock = /var/tmp/mtest-20700/.s.monetdb.37762
# cmdline opt 	monet_daemon = yes
# cmdline opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-globimport/BUILD/var/MonetDB/mTests_sql_test_copy
#client1:!ERROR: MT_glob_start: no files matched the pattern
#client1:!ERROR:ParseException:SQLparser:42000!COPY INTO: no file matched the expression: "/home/ferreira/repositories/monetdb-test-data/netflix_data/idontexist*"
MAPI  = (monetdb) /var/tmp/mtest-20700/.s.monetdb.37762
QUERY = copy into "ratings" from '/home/ferreira/repositories/monetdb-test-data/netflix_data/idontexist*' using delimiters ',','
        '; --error
ERROR = !COPY INTO: no file matched the expression: "/home/ferreira/repositories/monetdb-test-data/netflix_data/idontexist*"
CODE  = 42000

# 10:27:35 >  
# 10:27:35 >  "Done."
# 10:27:35 >  

