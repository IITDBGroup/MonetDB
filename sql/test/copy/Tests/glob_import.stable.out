stdout of test 'glob_import` in directory 'sql/test/copy` itself:


# 10:27:34 >  
# 10:27:34 >  "/usr/bin/python3" "glob_import.py" "glob_import"
# 10:27:34 >  

# MonetDB 5 server v11.34.0 (hg id: 46ba7db7f1bd)
# This is an unreleased version
# Serving database 'mTests_sql_test_copy', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.527 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2019 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://localhost.localdomain:37762/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-20700/.s.monetdb.37762
# MonetDB/GIS module loaded
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_views.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_geom.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_profiler.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 60_wlcr.sql
# loading sql script: 72_fits.sql
# loading sql script: 74_netcdf.sql
# loading sql script: 75_lidar.sql
# loading sql script: 75_shp.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 85_bam.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql
# MonetDB/SQL module loaded
#start transaction;
#create table "ratings" (movie_id BIGINT, customer_id BIGINT, rating TINYINT, rating_date DATE);
#copy into "ratings" from '/home/ferreira/repositories/monetdb-test-data/netflix_data/ratings_sample_0.csv' using delimiters ',','
#';
[ 1000	]
#copy into "ratings" from '/home/ferreira/repositories/monetdb-test-data/netflix_data/ratings_sample_1.csv' using delimiters ',','
#';
[ 1000	]
#copy into "ratings" from '/home/ferreira/repositories/monetdb-test-data/netflix_data/ratings_sample_?.csv' using delimiters ',','
#';
[ 2000	]
#copy into "ratings" from '/home/ferreira/repositories/monetdb-test-data/netflix_data/ratings_sample_*' using delimiters ',','
#';
[ 4000	]
#rollback;

# 10:27:35 >  
# 10:27:35 >  "Done."
# 10:27:35 >  

